cpp/include/cudf/strings/detail/gather.cuh

```diff
@@ -315,7 +315,7 @@ std::unique_ptr<cudf::column> gather(
     d_out_offsets + output_count,
     [] __device__(auto size) { return static_cast<size_t>(size); },
     size_t{0},
-    thrust::plus<size_t>{});
+    thrust::plus{});
```

Comment:
This is [surprising](https://ahseeit.com/meme-templates/king-include/uploads/2020/10/121135004_667644147516619_5170992311525198910_n-190849628.jpg) to me. 
How does it deduce the type without any parameters?

Reply:
Me too. Does it somehow use the other arguments to xform_reduce?

Reply:
The return type is dictated by the type of the initial value, aka `size_t{0}` on the line above.